datafusion/expr/src/expr.rs

```diff
@@ -983,10 +983,290 @@ impl PartialOrd for Expr {
 }
 
 impl Expr {
-    /// Returns the name of this expression as it should appear in a schema. This name
```

Comment:
Since this is a pub function what do you think about leaving it in marked as deprecated to help migration?

I think this make the PR much easier to handle

Something like

```rust
    #[deprecated(since = "31.0.0", note = "use data_type instead")]
    pub fn display_name(&self) -> Result<String> {
      self.schema_name()
    }
```